driver/src/main/java/org/kaazing/k3po/driver/internal/control/handler/ControlServerHandler.java

```diff
@@ -106,6 +109,28 @@ public void prepareReceived(final ChannelHandlerContext ctx, MessageEvent evt) t
             return;
         }
 
+        if ((lastRobot != null && ! lastRobot.getDisposedFuture().isDone()) || ! testFinished.compareAndSet(true, false)) {
```

Comment:
I think it is better the way it was before without the testFinished.compareAndSet.  Seems to be added complexity.  Alternatively, you could reduce the lastRobot and testFinished to a single variable most likely with a a queue